### PR TITLE
Pseudo methods in DocComments support

### PR DESCRIPTION
**Pull request by miguelbacalhau**
Pseudo properties were covered but not the methods. 

Not only the the specific class methods but also the parents since they are inherited (the same applies to the properties, so I went ahead and added it as well). 

The functionality is very similar to the pseudo properties one mostly differences in the regex pattern.

Some additional verifications and return values were added as well to keep this running smooth.

### CODE REVIEW

File: php/PHPCD.php [hunk 1]
```diff
@@ -211,17 +211,29 @@ private function docClass($class_name, $name, $is_method)
...
 
         if ($is_method) {
-            $reflection = $reflection_class->getMethod($name);
+            if ($reflection_class->hasMethod($name)) {
+                $reflection = $reflection_class->getMethod($name);
+            } else {
+                $class_doc = $this->getAllClassDocComments($reflection_class);
+                $has_pseudo_method = preg_match('/@method\s+(?<type>\S+)\s+?'.$name.'/mi', $class_doc, $matches);
+                if ($has_pseudo_method) {
+                    return [$reflection_class->getFileName(), '@var '.$matches['type']];
```
Comment (lvht) [inline php/PHPCD.php]: the returned file path should be the file path declaring the pseudo method doc

**Comment by miguelbacalhau** [comment]
@lvht I received an email about a comment that does not seem to be here. Anyway I changed the logic a bit to take into account the file name of the class that as the pseudo method.

**Comment by lvht** [comment]
I tried you patch this morning. It has some error. However, I am busy on work these days. Please give me more time to try you patch.

**Comment by miguelbacalhau** [comment]
Take the time you need.

The patch was not based on master before(it is now), that must have been the cause for problems.

**Comment by miguelbacalhau** [comment]
@lvht have you had a chance to try the patch?

It should still be okay if there is any problem just try to rebase it on top of master.

**Comment by lvht** [comment]
@miguelbacalhau you said that this patch is not based on the master branch. However, I cannot see any changes after your last comment.

**Comment by miguelbacalhau** [comment]
It is based on master I only said if there is any issues try to rebase it and see if there is still any issues.

But it should be okay.

**Comment by lvht** [comment]
ok. I will make my best effort to test this patch. :)

**Pull request merged**